### PR TITLE
Added User in EventDB

### PR DESCRIPTION
**Pull request by dhruv0000**
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a branch at [0chain/system_test](https://github.com/0chain/system_test) then add and merge your test changes to master BEFORE merging this PR

### CODE REVIEW

File: code/go/0chain.net/chaincore/chain/state_handler.go [hunk 1]
```diff
@@ -115,6 +115,26 @@ func (c *Chain) GetBalanceHandler(ctx context.Context, r *http.Request) (interfa
 	return state, nil
 }
 
+/*GetBalanceHandler - get the balance of a client */
+func (c *Chain) GetBalanceHandler(ctx context.Context, r *http.Request) (interface{}, error) {
+	clientID := r.FormValue("client_id")
+
+	evnetDb := c.GetEventDb()
```
Comment (kushthedude) [inline code/go/0chain.net/chaincore/chain/state_handler.go:122]: Typo

File: code/go/0chain.net/chaincore/chain/state_handler.go [hunk 2]
```diff
@@ -115,6 +115,26 @@ func (c *Chain) GetBalanceHandler(ctx context.Context, r *http.Request) (interfa
...
+	if evnetDb == nil {
+		return c.Deprecated_GetBalanceHandler(ctx, r)
+	}
+
+	user, err := evnetDb.GetUser(clientID)
+	if err != nil {
+		return c.Deprecated_GetBalanceHandler(ctx, r)
+	}
```
Comment (kushthedude) [inline code/go/0chain.net/chaincore/chain/state_handler.go:130]: Why is this needed?

File: code/go/0chain.net/chaincore/chain/state_handler.go [hunk 3]
```diff
@@ -115,6 +115,26 @@ func (c *Chain) GetBalanceHandler(ctx context.Context, r *http.Request) (interfa
...
+
+	// Create State Object from user's transiction hash?
```
Comment (kushthedude) [inline code/go/0chain.net/chaincore/chain/state_handler.go:132]: Where is the event being emmitted?
```
Change balance events will need to be emitted each time a user's wallet balance changes. Changes to:

AddTransfer
AddMint.
```

File: code/go/0chain.net/smartcontract/dbs/event/user.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+type User struct {
+	gorm.Model
+	// Dependency: PR #759. (Default name of the forign key is `transactionID` unless struct name is changed.)
+	TransactionID string `json:"TransactionID" gorm:"transactionID"`
+
+	UserID  string        `json:"UserID" gorm:"userID"`
+	Balance state.Balance `json:"Balance" gorm:"balance"`
+	// Dependency: PR #765. (Change int64 to Nounce type.)
+	Nonce int64 `json:"Nonce" gorm:"nonce"`
```
Comment (kushthedude) [inline code/go/0chain.net/smartcontract/dbs/event/user.go:18]: Not very much fluent with gorm but dont we need special annotations on foreign & primary key?

**Comment by dhruv0000** [comment]
Updated PR at https://github.com/0chain/0chain/pull/864. Closing this.
